aten/src/ATen/native/Loss.cpp

```diff
@@ -41,20 +41,18 @@ Tensor cosine_embedding_loss(const Tensor& input1, const Tensor& input2, const T
   auto denom = (mag_square1 * mag_square2).sqrt_();
   auto cos = prod_sum / denom;
 
-  auto zeros = at::zeros_like(cos, LEGACY_CONTIGUOUS_MEMORY_FORMAT);
   auto pos = 1 - cos;
   auto neg = (cos - margin).clamp_min_(0);
-  auto output_pos = at::where(target == 1, pos, zeros);
-  auto output_neg = at::where(target == -1, neg, zeros);
+  auto output_pos = at::where(target == 1, pos, 0);
+  auto output_neg = at::where(target == -1, neg, 0);
   auto output = output_pos + output_neg;
   return apply_loss_reduction(output, reduction);
 }
 
 Tensor hinge_embedding_loss(const Tensor& self, const Tensor& target, double margin, int64_t reduction) {
-  auto zeros = at::zeros_like(self, LEGACY_CONTIGUOUS_MEMORY_FORMAT);
   auto margin_clamp = (margin - self).clamp_min_(0);
-  auto output_margin = at::where(target != 1, margin_clamp, zeros);
-  auto output_self = at::where(target != -1, self, zeros);
+  auto output_margin = at::where(target != 1, margin_clamp, 0);
```

Comment:
will there not be problem because of int scalar type? margin_clamp has probably float tensor type
or will aten promote types here?

Reply:
At the moment, if there is one `scalar` and one `tensor`, the `scalar`'s dtype is cast to the type of `tensor`, this felt natural when implementing. But looking back, I guess this isn't a good behaviour, as it may lead to promotion and demotion for the type of `scalar` based on `tensor`.

Reply:
Ah, cool. So there is type promotion scalar -> tensor. For me that's what I would expect